jobsdb/jobsdb.go

```diff
@@ -4536,11 +4555,22 @@ func (jd *HandleT) GetProcessed(ctx context.Context, params GetQueryParamsT) (Jo
 	}
 
 	var completeProcessedJobs JobsResult
+	dsQueryCount := 0
+	var dsLimit int
+	if jd.dsLimit != nil {
+		dsLimit = *jd.dsLimit
+	}
 	for _, ds := range dsList {
-		processedJobs, err := jd.getProcessedJobsDS(ctx, ds, false, params)
+		if dsLimit > 0 && dsQueryCount >= dsLimit {
+			break
+		}
+		processedJobs, dsHit, err := jd.getProcessedJobsDS(ctx, ds, false, params)
 		if err != nil {
 			return JobsResult{}, err
 		}
+		if dsHit {
+			dsQueryCount++
+		}
```

Comment:
We only increase `dsQueryCount` when a dsHit takes place, and the decision to stop is based on that. 

What if dsQueryCount:
1.  was measured regardless of queries hit ?
2. we stopped early once the dsQueryCount was reached and we had non-zero data.

Letting 

The current approach makes sense, but I feel it doesn't penalise queries early enough when we have sparse entries.

Assume we have 200 tables and data are only on in table 3:
* first one
* one in the middle 
* one in the end 

Now, this algorithm will scan all 200 tables twice until the cache kicks in and avoid the lookup in some tables.

With the approaches I am suggesting, it will require 5 queries for the tables to be scanned, but those queries will be bounded.

Reply:
@lvrach a `dsHit` happens whenever we query the dataset, regardless if the query returns results or not.
The only occasions where `dsHit` will be returned as `false` are:
1. If we have a cache hit (no jobs & no query)
2. An error is returned for whatever reason

Reply:
Got it, my bad it wasn't clear from the diff. 

A small comment about what `getProcessedJobsDS` returns would be useful here.